yoti-sdk-auth/src/main/java/com/yoti/auth/AuthenticationTokenGenerator.java

```diff
@@ -0,0 +1,166 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+
+/**
+ * The {@link AuthenticationTokenGenerator} is used for generation authorization tokens
+ * that can be used for accessing Yoti services.  An authorization token must have
+ * a unique identifier, and an expiry timestamp.  One or more scopes can be provided
+ * to allow the authorization token access to different parts of Yoti systems.
+ * <p>
+ * The {@link AuthenticationTokenGenerator.Builder} can be accessed via {@code AuthorizationTokenGenerator.builder()}
+ * method, and then configured via the fluent API.
+ */
+public class AuthenticationTokenGenerator {
+
+    private static final Map<String, String> DEFAULT_FORM_VALUES;
+
+    static {
+        DEFAULT_FORM_VALUES = new HashMap<>();
+        DEFAULT_FORM_VALUES.put("grant_type", "client_credentials");
+        DEFAULT_FORM_VALUES.put("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
+    }
+
+    private final String sdkId;
+    private final KeyPair keyPair;
+    private final Supplier<String> jwtIdSupplier;
+    private final FormRequestClient formRequestClient;
+    private final SignedJwtGenerator signedJwtGenerator;
+
+    private final String authApiUrl;
+    private final ObjectMapper objectMapper;
+
+    AuthenticationTokenGenerator(
+            String sdkId,
+            KeyPair keyPair,
+            Supplier<String> jwtIdSupplier,
+            FormRequestClient formRequestClient,
+            SignedJwtGenerator signedJwtGenerator,
+            ObjectMapper objectMapper) {
+        this.sdkId = sdkId;
+        this.keyPair = keyPair;
+        this.jwtIdSupplier = jwtIdSupplier;
+        this.formRequestClient = formRequestClient;
+        this.signedJwtGenerator = signedJwtGenerator;
+        this.objectMapper = objectMapper;
+
+        authApiUrl = System.getProperty(Properties.PROPERTY_YOTI_AUTH_URL, Properties.DEFAULT_YOTI_AUTH_URL);
+    }
+
+    /**
+     * Creates a new instance of {@link AuthenticationTokenGenerator.Builder}
+     *
+     * @return the builder
+     */
+    public static AuthenticationTokenGenerator.Builder builder() {
+        return new AuthenticationTokenGenerator.Builder();
+    }
+
+    /**
+     * Creates a new authentication token, using the supplied scopes and comment.
+     *
+     * @param scopes  a list of scopes to be used by the authentication token
+     * @return a {@link CreateAuthenticationTokenResponse} containing information about the created token.
+     * @throws ResourceException if something was incorrect with the request to the Yoti authentication service
+     * @throws IOException
+     */
+    public CreateAuthenticationTokenResponse generate(List<String> scopes) throws ResourceException, IOException {
+        String jwts = signedJwtGenerator.create(sdkId, keyPair, jwtIdSupplier, authApiUrl);
+
+        Map<String, String> formParams = new HashMap<>(DEFAULT_FORM_VALUES);
```

Comment:
Is it really worth having a constant for `DEFAULT_FORM_VALUES`? It's only used here, why not inline the `put` here?
```java
formParams.put("scope", String.join(" ", scopes));
formParams.put("client_assertion", jwts);
formParams.put("grant_type", "client_credentials");
formParams.put("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
```
it's one less map instance to keep/copy and initialise on startup.